vortex-array/src/array/struct_/compute/to_arrow.rs

```diff
@@ -24,6 +26,8 @@ impl ToArrowFn<StructArray> for StructEncoding {
             .iter()
             .zip_eq(array.children())
             .map(|(field, arr)| {
+                let target_dtype = DType::from_arrow(field.as_ref());
+                let arr = try_cast(arr, &target_dtype)?;
```

Comment:
Can you add one more check, `target_dtype.eq_ignore_nullability(arr.dtype())` - just to make sure we're _only_ doing null casting?

And leave a comment as to why